students/Dustin_L/lesson04/mailroom2.py

```diff
@@ -0,0 +1,219 @@
+                                       AVE_KEY: 0}})
+
+THANK_YOU_FMT = ('\nDear {:s},\n'
+                 'Thank you for your generous donation of ${:.2f}.\n'
+                 '\t\tSincerely,\n'
+                 '\t\t  -Your conscience')
+
+
+def get_usr_input():
+    """Get input from user.
+
+    Prompt user to select one of three choices. If the user selects one of
+    these three, that value is returned. If not, the user is prompted again to
+    select.
+
+    Returns:
+        int: Value corresponding to user choice
+    """
+    select_prompt = ('\nPlease select from the following options:\n'
+                     '\t1. Send a Thank You\n'
+                     '\t2. Create a Report\n'
+                     '\t3. Send letters to all donors\n'
+                     '\t4. quit\n'
+                     ' --> ')
+    usr_in = int(input(select_prompt))
+
+    while usr_in not in (THANK_YOU_OPT, REPORT_OPT, LETTERS_OPT, QUIT_OPT):
+        print('\nPlease enter either a "1", "2", or "3"')
+        usr_in = int(input(select_prompt))
+
+    return usr_in
+
+
+def add_donation(donor, amount):
+    """Add a new donation to the donation database.
+
+    Args:
+        donor (str): Name of donor in donation database.
+        amount (int): Amount to add to donation database.
+    """
+    DONOR_DB[donor][GIFTS_KEY].append(amount)
+    DONOR_DB[donor][NUM_GIFTS_KEY] += 1
+    DONOR_DB[donor][TOTAL_KEY] += amount
+    DONOR_DB[donor][AVE_KEY] = DONOR_DB[donor][TOTAL_KEY] / \
+        DONOR_DB[donor][NUM_GIFTS_KEY]
+
+
+def send_thank_you():
+    """Send a thank you.
+
+    Prompt for a Full Name.
+    If the user types ‘list’, show them a list of the donor names and re-prompt
+    If the user types a name not in the list, add that name to the data
+    structure and use it.
+    If the user types a name in the list, use it.
+    Once a name has been selected, prompt for a donation amount.
+    Turn the amount into a number – it is OK at this point for the program to
+    crash if someone types a bogus amount.
+    Once an amount has been given, add that amount to the donation history of
+    the selected user.
+    Finally, use string formatting to compose an email thanking the donor for
+    their generous donation. Print the email to the terminal and return to the
+    original prompt.
+    """
+    name_prompt = '\nPlease enter name of "Thank You" recipient:\n'\
+                  '(Enter "list" to see all donors)\n'\
+                  '(Enter "quit" to return to main menu)\n'\
+                  ' --> '
+    amount_prompt = '\nPlease enter the donation amount:\n'\
+                    '(Enter "quit" to return to main menu)\n'\
+                    ' --> '
+    names = [donor.lower() for donor in DONOR_DB]
+
+    while True:
+        usr_in = input(name_prompt).strip().lower()
+
+        if usr_in.startswith('q'):
+            break
+        elif usr_in == 'list':
+            print()
+            [print(name.title()) for name in names]
```

Comment:
for your list comprehension question, this is certainly valid but I wouldn't say this is the best practice.

list comprehensions are meant to return a list and so here you are wastefully generating another list and only using it to make it one-liner.

if you would like to keep code cleaner you can always extract a piece of code into a function, but given how simple this part is I do not know that it is worth it.